src/js/components/projects/detail.tsx

```diff
@@ -21,13 +26,60 @@ import {
   useFetchTasksIfMissing,
 } from '@/components/utils';
 import SubmitModal from '@/components/utils/submitModal';
+import { ThunkDispatch } from '@/store';
+import { setUsersOnProject } from '@/store/projects/actions';
+import { GitHubUser } from '@/store/repositories/reducer';
 import routes from '@/utils/routes';
 
 const ProjectDetail = (props: RouteComponentProps) => {
+  const dispatch = useDispatch<ThunkDispatch>();
   const { repository, repositorySlug } = useFetchRepositoryIfMissing(props);
   const { project, projectSlug } = useFetchProjectIfMissing(repository, props);
   const { tasks } = useFetchTasksIfMissing(project, props);
 
+  // Assign users modal related:
+  const [assignUsersModalOpen, setAssignUsersModalOpen] = useState(false);
+  const openAvailableUserModal = () => {
+    setAssignUsersModalOpen(true);
+  };
+  const closeAvailableUserModal = () => {
+    setAssignUsersModalOpen(false);
+  };
+  const setProjectUsers = useCallback(
+    (users: GitHubUser[]) => {
+      if (!project) {
+        return;
+      }
+      dispatch(
+        setUsersOnProject({
+          ...project,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          github_users: users,
+        }),
+      );
+      setAssignUsersModalOpen(false);
+    },
+    [project, dispatch],
+  );
+  const removeUser = useCallback(
+    (user: GitHubUser) => {
+      if (!project) {
+        return;
+      }
+      const users = project.github_users.filter(
+        (possibleUser: GitHubUser) => user !== possibleUser,
```

Comment:
Comparing objects like this in JS is fragile.

```suggestion
        (possibleUser: GitHubUser) => user.id !== possibleUser.id,
```